### PR TITLE
Cleanup logs

### PR DESCRIPTION
**Pull request by SwampDragons**
While working on documentation for our rpc servers and hooks, I found several places where the logs could be improved with a little TLC. Each change gets its own explanatory commit.

### CODE REVIEW

File: packer/plugin/server.go [hunk 1]
```diff
@@ -104,15 +91,28 @@ func Server() (*packrpc.Server, error) {
...
+		return nil, err
+	}
+
+	maxPort, err := strconv.ParseInt(os.Getenv("PACKER_PLUGIN_MAX_PORT"), 10, 32)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Plugin minimum port: %d\n", minPort)
+	log.Printf("Plugin maximum port: %d\n", maxPort)
```
Comment (azr) [inline packer/plugin/server.go]: Would it be a good idea to regroup these two lines into one ? Like I did in #8423 ?

File: config.go [hunk 2]
```diff
@@ -208,29 +208,32 @@ func (c *config) discoverInternal() error {
 	for builder := range command.Builders {
 		_, found := (c.Builders)[builder]
 		if !found {
-			log.Printf("Using internal plugin for %s", builder)
 			(c.Builders)[builder] = fmt.Sprintf("%s%splugin%spacker-builder-%s",
 				packerPath, PACKERSPACE, PACKERSPACE, builder)
+		} else {
+			log.Printf("Using external plugin for %s", builder)
```
Comment (azr) [inline config.go]: Nice, yeah; this will be way less verbose; updating my branch !

File: packer/plugin/client.go [hunk 3]
```diff
@@ -339,14 +341,22 @@ func (c *Client) Start() (addr net.Addr, err error) {
 }
 
 func (c *Client) logStderr(r io.Reader) {
+	logPrefix := filepath.Base(c.config.Cmd.Path)
+	if logPrefix == "packer" {
+		// we just called the normal packer binary with the plugin arg.
+		// grab the last arg from the list which will match the plugin name.
+		logPrefix = c.config.Cmd.Args[len(c.config.Cmd.Args)-1]
```
Comment (azr) [inline packer/plugin/client.go]: awesome !

File: packer/plugin/server.go (hunk 1 above)
Comment (SwampDragons) [reply packer/plugin/server.go]: Yeah, I see no reason not to.

**Pull request merged**